Marlin/gcode.h

```diff
@@ -148,8 +147,7 @@ class GCodeParser {
 
     // Code is found in the string. If not found, value_ptr is unchanged.
     // This allows "if (seen('A')||seen('B'))" to use the last-found value.
-    // This is volatile because its side-effects are important
-    static volatile bool seen(const char c) {
+    static bool seen(const char& c) {
```

Comment:
Ah, I see. `volatile` is connected to the potential of member vars to be altered, not just to the concept of non-optimization. The solution is to remove the `static` keyword.